misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractIterableNotToContainEntriesExpectationsTest.kt

```diff
@@ -0,0 +1,294 @@
+import ch.tutteli.atrium.core.polyfills.format
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.logic.utils.expectLambda
+import ch.tutteli.atrium.specs.Fun2
+import ch.tutteli.atrium.specs.explanatoryBulletPoint
+import ch.tutteli.atrium.specs.forSubjectLessTest
+import ch.tutteli.atrium.specs.hintBulletPoint
+import ch.tutteli.atrium.specs.indentListBulletPoint
+import ch.tutteli.atrium.specs.indentRootBulletPoint
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.afterExplanatory
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.afterExplanatoryIndent
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.afterMismatchedWarning
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.anElementWhichNeedsDescr
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.hasANextElement
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.index
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.mismatchedIndex
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.toBeGreaterThanFun
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.toBeLessThanFun
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.toEqualFun
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.mismatches
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.oneToSeven
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.oneToSevenNullable
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.separator
+import ch.tutteli.atrium.specs.integration.utils.ExpectationCreatorTestData
+import ch.tutteli.atrium.specs.integration.utils.ExpectationCreatorTriple
+import ch.tutteli.atrium.specs.integration.utils.SubjectLessTestData
+import ch.tutteli.atrium.specs.invoke
+import ch.tutteli.atrium.specs.listBulletPoint
+import ch.tutteli.atrium.specs.name
+import ch.tutteli.atrium.specs.rootBulletPoint
+import ch.tutteli.atrium.specs.toEqualDescr
+import ch.tutteli.atrium.specs.warningBulletPoint
+import ch.tutteli.atrium.testfactories.TestFactory
+import ch.tutteli.atrium.translations.DescriptionComparableExpectation
+import ch.tutteli.atrium.translations.DescriptionIterableLikeExpectation
+
+
+@Suppress("FunctionName")
+abstract class AbstractIterableNotToContainEntriesExpectationsTest(
+    private val notToContainEntriesSpec: Fun2<Iterable<Double>, Expect<Double>.() -> Unit, Array<out Expect<Double>.() -> Unit>>,
+    private val notToContainNullableEntriesSpec: Fun2<Iterable<Double?>, (Expect<Double>.() -> Unit)?, Array<out (Expect<Double>.() -> Unit)?>>,
+    private val notToHaveElementsOrNoneFunNameSpec: String,
```

Comment:
Please rename to the following (is not a spec):
```suggestion
    private val notToHaveElementsOrNoneFunName: String,
```